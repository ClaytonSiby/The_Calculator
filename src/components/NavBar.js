import React from 'react'
import { BrowserRouter as Router, Route, Switch, Link } from 'react-router-dom'
import App from './App'
import Home from './Home'
import Quote from './Quote'

const NavBar = () => {
  return (
    <Router>
      <nav>
        <div>
            <h2>Math Magicians</h2>
        </div>
        <ul>
          <li>
            <Link to='/'>Home</Link>
          </li>
          <li>
            <Link to='/calculator'>Calculator</Link>
          </li>
          <li>
            <Link to='/quote'>Quote</Link>
          </li>
        </ul>
      </nav>

      <Switch>
        <Route exact path='/'>
          <Home />
        </Route>
        <Route path='/calculator'>
          <App />
        </Route>
        <Route path='/quote'>
          <Quote />
        </Route>
      </Switch>
    </Router>
  )
}

export default NavBar
