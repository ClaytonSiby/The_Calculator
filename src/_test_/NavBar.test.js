import renderer from 'react-test-renderer';
import NavBar from '../components/NavBar';

describe('NavBar component', () => {
  test('render successfully to the DOM', () => {
    const test = renderer.create(<NavBar />).toJSON();
    expect(test).toMatchSnapshot();
  });
});
