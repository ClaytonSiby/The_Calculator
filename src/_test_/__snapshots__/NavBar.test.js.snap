// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component render successfully to the DOM 1`] = `
Array [
  <nav>
    <div>
      <h2>
        Math Magicians
      </h2>
    </div>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="homeContainer"
  >
    <div>
      <p>
        Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.
      </p>
      <p>
        We will always have STEM with us. Some things will drop out of the public eye and go away, but there will always be science, engineering and technology. And there will always, always be mathematics.
      </p>
    </div>
  </div>,
]
`;
